### PR TITLE
Update to use use new vagrant images

### PR DESCRIPTION
**Pull request by aahmed-se**
We are updating the osx build to use new optimized vm images with pre installed dev tools , brew and associated build packages. The calling script can now link the python and boost version required and generated the mac wheel packages. The VM images have also been  reduced in size by removing unused applications.

### CODE REVIEW

**Comment by sijie** [comment]
@merlimat can you review this?

File: pulsar-client-cpp/python/pkg/osx/osx-10.12-python2.7/Vagrantfile [hunk 1]
```diff
@@ -20,8 +20,9 @@
 #
 
 Vagrant.configure("2") do |config|
-  config.vm.box = "jhcook/macos-sierra"
+  config.vm.box = "apachepulsare/osx-10.12-python-dev"
```
Comment (merlimat) [inline pulsar-client-cpp/python/pkg/osx/osx-10.12-python2.7/Vagrantfile]: `apachepulsare` -> `apachepulsar`

File: pulsar-client-cpp/python/pkg/osx/osx-10.13-python2.7/Vagrantfile [hunk 2]
```diff
@@ -20,9 +20,10 @@
 #
```
Comment (merlimat) [inline pulsar-client-cpp/python/pkg/osx/osx-10.13-python2.7/Vagrantfile]: Can you remove the 10.11 if they're not used

Comment (aahmed-se) [reply pulsar-client-cpp/python/pkg/osx/osx-10.13-python2.7/Vagrantfile]: it's already removed.

File: pulsar-client-cpp/python/pkg/osx/osx-10.12-python2.7/Vagrantfile (hunk 1 above)
Comment (aahmed-se) [reply pulsar-client-cpp/python/pkg/osx/osx-10.12-python2.7/Vagrantfile]: done.

**Pull request closed without merging**